### PR TITLE
Fix fileOffset in ParquetRowReader::filterRowGroups()

### PR DESCRIPTION
**Pull request by yingsu00**
ParquetRowReader::filterRowGroups() get the offset of the first data or dictionary page to the beginning of the file and assign it to fileOffset, which would be used to see if this RowGroup is in the split's range. It uses the ColumnChunk.file_offset (rowGroups_[i[.columns[0].file_offset) as the fileOffset when the Rowgroup's file_offset
(rowGroups_[i].file_offset) was not set, but ColumnChunk.file_offset is actually the offset to the ColumnMetaData, not the offset to the data or dictionary pages. Furthermore ColumnChunk.file_offset is optional, which would cause queries to fail. This commit fixed the fileOffset calculation to use ColumnMetaData.data_page_offset or ColumnMetaData.dictionary_page_offset, whichever is first to occur. The ColumnMetaData.data_page_offset is guaranteed to be existing.

### CODE REVIEW

**Comment by Yuhta** [comment]
We don't need #5118 once this is fixed, right?  Can you copy the test from #5118 here to make sure it is passing?

**Comment by yingsu00** [comment]
@Yuhta Thanks for reviewing so quickly. Updated the PR with the test. 
> We don't need https://github.com/facebookincubator/velox/pull/5118 once this is merged, right?

Correct.
> Can you copy the test from https://github.com/facebookincubator/velox/pull/5118 here to make sure it is passing?

Yes renamed it to filterRowGroups though.

File: velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp [hunk 1]
```diff
@@ -289,3 +289,21 @@ TEST_F(ParquetReaderTest, intMultipleFilters) {
...
+
+// This test is to verify filterRowGroups doesn't throw the fileOffset Velox
+// check failure
+TEST_F(ParquetReaderTest, filterRowGroups) {
+  // decimal_dict.parquet one columns a: DECIMAL(9,1)
+  // rowGroups_[0].columns[0].file_offset is 0.
+  auto rowType = ROW({"_c0"}, {DECIMAL(9, 1)});
+  ReaderOptions readerOpts{defaultPool.get()};
+  const std::string decimal_dict(
+      getExampleFilePath("decimal_zero_offset.parquet"));
```
Comment (Yuhta) [inline velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp:301]: The parquet file also need to be copied

**Comment by yingsu00** [comment]
@Yuhta HI Jimmy I updated the PR with the file. Can you please review again?

**Pull request closed without merging**